mccn/mccn.py

```diff
@@ -39,14 +39,26 @@ def load_stac(self, col_id: str, bands: Optional[List[str]] = None, groupby: str
             chunks = {'x': 2048, 'y': 2048}
         else:
             chunks = None
+        # TODO: We likely need a unique load function here depending on data type. For example
+        # TODO: we need to write at least one for raster data and time series data.
+
         xx = stac_load(
             self._query(col_id), bands=bands, groupby=groupby, chunks=chunks,  # type: ignore
             progress=tqdm, pool=pool, crs=crs, geobox=geobox
         )
-        min_lon = xx.longitude.min().item()
-        max_lon = xx.longitude.max().item()
-        min_lat = xx.latitude.min().item()
-        max_lat = xx.latitude.max().item()
+        if 'x' in xx.xindexes and 'y' in xx.xindexes:
```

Comment:
I suppose we could leave some info in the stac generator about x/y name, probably in the properties dict

Reply:
its a function of the epsg code but not sure how to calculate it on the fly.